Allowing override LastUpgradeCheck and LastUpgradeVersion

Andre113 opened the pull request:
Xcode will check for the value of `LastUpgradeCheck` and `LastUpgradeVersion` to display some warnings related to a recommended iOS deployment target. Just like this:

<img src="https://user-images.githubusercontent.com/10833549/106525657-134ac680-64c3-11eb-824c-5ff7c69fc976.png" width="40%" height="40%">

In some cases, it is not possible to change the deployment target because we still have to support some old iOS versions, and then the warnings stays there forever. This is really bad when we have a large number of targets.

With these changes, we can specify a different value for `LastUpgradeCheck` in `project.yml` . Like this:

```
name: MyApp

atributes:
  LastUpgradeCheck: '8.3'

configs:
  Debug: debug
  Homolog: debug
  Release: release
configFiles:
  Debug: Configs/DebugBase.xcconfig
  Homolog: Configs/HomologBase.xcconfig
  Release: Configs/ProdBase.xcconfig
```

This way, the warnings should be gone.

yonaskolb commented on Sources/XcodeGenKit/PBXProjGenerator.swift:
```diff
@@ -285,9 +285,16 @@ public class PBXProjGenerator {
+        if !projectAttributes.contains(where: { (key, value) -> Bool in
+            key == lastUpgradeKey && value is String
+        }) {
+            projectAttributes[lastUpgradeKey] = project.xcodeVersion
```
This code is the same as before, but it only replaces the key if it's a string, correct? Were you running into cases where devs were putting numbers in the project attributes?

Andre113 replied:
Almost. It will replace the key if the user did not specify a value for `LastUpgradeCheck` or if this value is not a string. In this case, it will keep the same behavior as today, using the `project.xcodeVersion` value.
I  just felt it would be safer to validade if the value we're inserting is a `String` to avoid any strange values.

yonaskolb commented:
Thanks @Andre113. Could you add a changelog entry as well

Andre113 commented:
> Thanks @Andre113. Could you add a changelog entry as well

Sure! Just updated it. I also modified the readme to update the information regard the `LastUpgradeCheck` property.

yonaskolb commented:
Thanks @Andre113. We’ve had a release since you added your change log entry so could you move it up into the Next Version section

Andre113 commented:
> Thanks @Andre113. We’ve had a release since you added your change log entry so could you move it up into the Next Version section

Oops.
Done, just updated it.
